src/cli/commands/files/add.js

```diff
@@ -55,44 +55,71 @@ module.exports = {
     const index = inPath.lastIndexOf('/') + 1
 
     utils.getIPFS((err, ipfs) => {
-      if (err) throw err
+      if (err) {
+        throw err
+      }
+
+      // pull-stream -> stream -> pull-stream
+      // is not a good idea, so if we have a pull interface
+      // available, use it
+      let createAddStream = (cb) => {
+        ipfs.files.createAddStream((err, stream) => {
+          cb(err, err ? null : toPull.transform(stream))
+        })
+      }
+
+      if (typeof ipfs.files.createAddPullStream === 'function') {
+        createAddStream = (cb) => {
+          cb(null, ipfs.files.createAddPullStream())
+        }
+      }
+
+      createAddStream((err, addStream) => {
+        if (err) {
+          throw err
+        }
+
+        glob(path.join(inPath, '/**/*'), (err, list) => {
+          if (err) throw err
+          if (list.length === 0) list = [inPath]
```

Comment:
Please make a PR to aegir to enforce this if you are going to comment on every single curly